skonfig/configuration.py

```diff
@@ -20,48 +17,63 @@ def _set_defaults(configuration):
             configuration[option] = defaults[option]
 
 
-def _set_conf_dirs(configuration):
-    # loading order:
-    #   ~/.skonfig/set/
-    #   ~/.skonfig/config:conf_dirs = ...
-    #   ~/.skonfig/
-    if "conf_dir" in configuration:
-        configuration["conf_dir"] = [
-            conf_dir
-            for conf_dir in configuration["conf_dir"].split(os.pathsep)
-            if os.path.isdir(conf_dir)
-        ]
-    else:
-        configuration["conf_dir"] = []
-    if os.path.isdir(CONFIGURATION_HOME_PATH):
-        configuration["conf_dir"].insert(0, CONFIGURATION_HOME_PATH)
-    # find sets and insert them into PATH
-    sets_dir = os.path.join(CONFIGURATION_HOME_PATH, "set")
-    if os.path.isdir(sets_dir):
-        configuration["conf_dir"] += [
-            os.path.join(sets_dir, s) for s in os.listdir(sets_dir)
-        ]
-    configuration["conf_dir"].reverse()
+def _get_search_dirs():
+    pathsep_search_dirs = os.getenv("SKONFIG_PATH", "~/.skonfig:/etc/skonfig")
+    search_dirs = []
+    for search_dir in pathsep_search_dirs.split(":"):
+        search_dir = os.path.expanduser(search_dir)
+        if not os.path.isdir(search_dir):
+            continue
+        search_dirs.append(search_dir)
+    search_dirs.reverse()
```

Comment:
This is how `cdist` core behaves (last wins). This way we can define things as we normally would with `$PATH` and then serve it reversed to `cdist` core. I'll add comment about that.

And I'm really in between with `.`. `git` does it, why wouldn't we?

Reply:
And it broke :sunglasses: 

When I'm in type directory and it contains `files` and `explorer` directories, then the contents of these directories will be linked to global counterparts:
```
TRACE: localhost: Linking /home/ander/.skonfig/type/__ander_fonts/explorer/console to /tmp/tmpxuo765eg/421aa90e079fa326b6494f812ad13e79/data/conf/explorer/console ... 
TRACE: localhost: Linking /home/ander/.skonfig/type/__ander_fonts/explorer/installed to /tmp/tmpxuo765eg/421aa90e079fa326b6494f812ad13e79/data/conf/explorer/installed ...         
TRACE: localhost: Linking /home/ander/.skonfig/type/__ander_fonts/files/packages to /tmp/tmpxuo765eg/421aa90e079fa326b6494f812ad13e79/data/conf/files/packages ...                       
TRACE: localhost: Linking /home/ander/.skonfig/type/__ander_fonts/files/segoe to /tmp/tmpxuo765eg/421aa90e079fa326b6494f812ad13e79/data/conf/files/segoe ...            
```

So, nope.

Reply:
I am not sure, I follow your thinking with reversing the list: are we reversing because the core reverses again; hence we end up with _first-one-wins_?

Regarding includein `.` - yet again, I am unsure what your point is: including the current directory would not work, if we're in the wrong directory?

Reply:
> are we reversing because the core reverses again; hence we end up with first-one-wins?

All the black magic happens here:
https://github.com/skonfig/skonfig/blob/64158959a55d7eaef25d73179ea892aa31b1f8b8/cdist/exec/local.py#L363

> including the current directory would not work, if we're in the wrong directory?

It works, but outcome can be bit weird.

For example if working directory is type directory and it has explorer with same name as global explorer. I don't know if this is something we want to avoid, but it feels bit slippery road for me.

Reply:
First of all, I did mess up my accounts and thus the comment history; long story short, the previous comment from @4nd3r  is in fact a response to this:

> I am not sure, I follow your thinking with reversing the list: are we reversing because the core reverses again; hence we end up with first-one-wins?
> 
> Regarding including . - yet again, I am unsure what your point is: including the current directory would not work, if we're in the wrong directory?

Moreover: Did you mean to link that source line with regards whether the first or the last one wins?

When it comes to `.` being in `$SKONFIG_PATH`, I can see where you're coming from. But then again, that seems to be very convenient. Also, I am not really convinced, the one-skonfig-dir-per-user approach would work for everyone.

I reckon, git would check, whether there's a `.git` directory, that would definitely answer the question of whether we're in the right place. Same with `manifest/init` with `skonfig` - but that feels to be a slippery slope too.

I guess, there's some thinking to be done here... is the only problem the `exploerer` subdirectory?

Reply:
> Moreover: Did you mean to link that source line with regards whether the first or the last one wins?

I linked location where order of conf dirs matter.

> I reckon, git would check, whether there's a .git directory, that would definitely answer the question of whether we're in the right place. Same with manifest/init with skonfig - but that feels to be a slippery slope too.
> I guess, there's some thinking to be done here...

I feel this question should be handled separately.

Reply:
Well, besides my pet-peeve around `.`, I think we're golden here...